systemtest/src/test/java/io/strimzi/systemtest/operators/LeaderElectionIsolatedST.java

```diff
@@ -0,0 +1,129 @@
+     */
+    final EnvVar leaderEnabledEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_ENABLED")
+        .withValue("true")
+        .build();
+
+    final EnvVar leaseNameEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_LEASE_NAME")
+        .withValue(Constants.STRIMZI_DEPLOYMENT_NAME)
+        .build();
+
+    final EnvVar leaseNamespaceEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_LEASE_NAMESPACE")
+        .withNewValueFrom()
+            .withNewFieldRef()
+                .withFieldPath("metadata.namespace")
+            .endFieldRef()
+        .endValueFrom()
+        .build();
+
+    final EnvVar leaderIdentityEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_IDENTITY")
+        .withNewValueFrom()
+            .withNewFieldRef()
+                .withFieldPath("metadata.name")
+            .endFieldRef()
+        .endValueFrom()
+        .build();
+
+    final List<EnvVar> envList = Arrays.asList(leaderEnabledEnv, leaseNameEnv, leaseNamespaceEnv, leaderIdentityEnv);
+
+    @IsolatedTest
+    void testLeaderElection(ExtensionContext extensionContext) {
+        TestStorage testStorage = new TestStorage(extensionContext);
+
+        clusterOperator = clusterOperator.defaultInstallation()
+            .withExtensionContext(extensionContext)
+            .withExtraEnvVars(envList)
+            .withReplicas(2)
+            .createInstallation()
+            .runInstallation();
+
+        Lease oldLease = kubeClient().getClient().leases().inNamespace(testStorage.getNamespaceName()).withName(Constants.STRIMZI_DEPLOYMENT_NAME).get();
+        String oldLeaderPodName = oldLease.getSpec().getHolderIdentity();
+
+        LOGGER.info("Changing image of the leader pod: {} to not available image - to cause CrashLoopBackOff and change of leader to second pod (failover)", oldLeaderPodName);
+
+        kubeClient().editPod(testStorage.getNamespaceName(), oldLeaderPodName).edit(pod -> new PodBuilder(pod)
+            .editOrNewSpec()
+                .editContainer(0)
+                    .withImage("wrong-image/name:latest")
+                .endContainer()
+            .endSpec()
+            .build()
+        );
+
+        PodUtils.waitUntilPodIsInCrashLoopBackOff(testStorage.getNamespaceName(), oldLeaderPodName);
+
+        Lease currentLease = kubeClient().getClient().leases().inNamespace(testStorage.getNamespaceName()).withName(Constants.STRIMZI_DEPLOYMENT_NAME).get();
+        String currentLeaderPodName = currentLease.getSpec().getHolderIdentity();
+
+        String logFromNewLeader = StUtils.getLogFromPodByTime(INFRA_NAMESPACE, currentLeaderPodName, Constants.STRIMZI_DEPLOYMENT_NAME, "300s");
+
+        assertTrue(logFromNewLeader.contains("I'm the new leader"));
+        assertNotEquals(oldLeaderPodName, currentLeaderPodName);
+    }
+
+    @BeforeAll
+    void setup() {
+        assumeTrue(!Environment.isOlmInstall());
```

Comment:
Adding some comment why OLM installation is not supported for this feature would be great.